src/models/state/wallet/wallet_state.rs

```diff
@@ -270,6 +298,33 @@ impl WalletState {
             .map(SpendingKey::RawHashLock)
             .collect_vec();
 
+        let scan_mode = match (&cli_args.scan_blocks, cli_args.scan_keys) {
+            (None, None) => {
+                if !wallet_was_new && database_was_new {
+                    info!("Activating scan mode: wallet file present but databse absent; wallet may have been imported.");
+                    Some(ScanModeConfiguration::default())
+                } else {
+                    None
+                }
+            }
```

Comment:
Question: Should this also take the presence of `incoming_randomness.dat` into account? If that file is present, do we need to scan then? Or should we account for it in any way?